src/main/java/org/apache/bcel/classfile/ClassParser.java

```diff
@@ -303,5 +304,6 @@ private void readMethods() throws IOException, ClassFormatException {
     private void readVersion() throws IOException, ClassFormatException {
         minor = dataInputStream.readUnsignedShort();
         major = dataInputStream.readUnsignedShort();
+        isOak = major < Const.MAJOR_1_1 || (major == Const.MAJOR_1_1 && minor < 3);
```

Comment:
Should the check be `minor <= 3` instead? See notes.

Reply:
> > [OakFile.java.zip](https://github.com/apache/commons-bcel/files/5054556/OakFile.java.zip)
> > There is no license, it's practically a hello world program.
> 
> Hi @Hippo
> I'll should be able to take a look this weekend to see if a cleaner approach can be devised. In the meantime, please provide the source .java and instructions on how you created the class file if it is out of the ordinary Java tooling we are all used to. Is there an Oak/Java 1.0 compiler you used we can download to create additional files?

No I don't have an oak compiler, but you can emulate one by transforming a class file and changing its version to a proper oak version and halfing the correct data types in the code attribute.

Reply:
> Should the check be `minor <= 3` instead? See notes.
> Are you saying that 45.2 is not compatible with 45.3?

No it should be `minor < 3` take a look at the jvm's class parser
https://github.com/AdoptOpenJDK/openjdk-jdk8u/blob/4b69552faa8de8f7de5bed1c95c12c025702b8ee/hotspot/src/share/vm/classfile/classFileParser.cpp#L2137